c/meterpreter/source/metsrv/remote_thread.c

```diff
@@ -79,7 +86,23 @@ HANDLE create_remote_thread(HANDLE hProcess, SIZE_T sStackSize, LPVOID pvStartAd
 
 			if (ntResult == 0 && pdwThreadId)
 			{
-				*pdwThreadId = GetThreadId(hThread);
+				if (!pGetThreadIdAttempted)
+				{
+					if (pGetThreadId == NULL)
+					{
+						pGetThreadId = (PGetThreadId)GetProcAddress(GetModuleHandleA("kernel32"), "GetThreadId");
+						if (pGetThreadId)
+						{
+							dprintf("[REMOTETHREAD] GetThreadId found at %p", pGetThreadId);
+						}
+					}
+					pGetThreadIdAttempted = TRUE;
+				}
+
+				if (pGetThreadId != NULL)
+					*pdwThreadId = pGetThreadId(hThread);
+				else
+					*pdwThreadId = 0;
```

Comment:
Note to reviewers/testers,  need to check if setting `*pdwThreadId` to 0 on systems that don't have `GetThreadId()` available will cause issues down the line.

Reply:
After a quick look at it, it looks like only [`base_inject.c`](https://github.com/rapid7/metasploit-payloads/blob/a86dc9db06a08f7724c9bc981cc8e71a076b4839/c/meterpreter/source/metsrv/base_inject.c#L430) calls `create_remote_thread()` and passes `NULL` as `pdwThreadId` argument. So, I think we're good.

Reply:
@cdelafuente-r7 technically speaking, there is no need to call GetThreadId, but it's enough to pass CIENT_ID param to RtlCreateUserThread. As output, the CLIENT_ID will hold UniqueThread value which is ThreadId, retrieved also by GetThreadId calling NtQueryInformationThread.

Reply:
I agree, I'm inclined to say simply avoid calling `GetThreadId` as you suggest.

Also, since this is not directly related to the original issue (https://github.com/rapid7/metasploit-payloads/issues/632), I'm wondering if this should be part of a separate PR, keeping only the fix for `OpenThreadToken` in this one.